components/ai_chat/distiller_scripts/scripts/x.com/utils.ts

```diff
@@ -0,0 +1,123 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { LEO_DISTILLATION_LEVEL } from '../distillation'
+import { getDistillationLevel } from './distiller'
+
+export function isString(value: any) {
+  return typeof value === 'string'
+}
+
+export function isSupportedPage(document: Document) {
+  const supportedPaths = [
+    '/',
+    '/home',
+    '/brave',
+    '/bravesampson',
+    '/brendaneich',
+    '/notifications',
+    /^\/.*?\/status\/\d+/ // Post detail pages
+  ]
+
+  for (const entry of supportedPaths) {
+    if (entry instanceof RegExp && entry.test(document.location.pathname)) {
+      return true
+    } else if (isString(entry) && entry === document.location.pathname) {
+      return true
+    }
+  }
+
+  return false
+}
+
+/**
+ * Decodes HTML entities in a given string using a
+ * `textarea` element.
+ */
+export function decodeHTMLSpecialChars(text: string) {
+  const textarea = document.createElement('textarea')
+  /**
+   * We have to use `.innerHTML` to decode entities, as
+   * `.innerText` or `.textContent` will not achieve
+   * the same results.
+   */
+  // eslint-disable-next-line no-unsanitized/property
+  textarea.innerHTML = text
```

Comment:
This is used for the purpose of rendering html special chars. Content is passed into the textarea, and then immediately read out through the static `value` property. No evals or execution anywhere. If there is a better approach to rendering special chars, I would be happy (quite eager, honestly) to change the approach 😀

Reply:
innerText not working here?
This might unescape stuff

Reply:
Unfortunately, innerText doesn't work here. That said, DOMParser to the rescue. I was a little worried about triggering requests, but DOMParser marks parsed-script tags as non-executable, etc. Hopefully this newer approach is more ideal.

Reply:
Having slept on it, I don't think DOMParser is the right approach either. For example, if a tweet/post contains an example HTML comment (e.g., "`<!-- this is a comment -->`"), the DOMParser will strip this from the output when we call `body.textContent`. If we were to call `body.innerHTML` instead, that would preserve HTML comments, but it would also alter the markup in many cases (e.g., altering examples of purposefully invalid markup, etc.). The only approach I can think of that works is the one-off use of `TEXTAREA.innerHTML`.

Reply:
@thypon, do you have any concerns with using `createHTMLDocument` instead of `createDocument`?